libraries/AP_InertialSensor/AP_InertialSensor.cpp

```diff
@@ -874,45 +876,46 @@ AP_InertialSensor::init(uint16_t loop_rate)
     // the center frequency of the harmonic notch is always taken from the calculated value so that it can be updated
     // dynamically, the calculated value is always some multiple of the configured center frequency, so start with the
     // configured value
-    _calculated_harmonic_notch_freq_hz[0] = _harmonic_notch_filter.center_freq_hz();
-    _num_calculated_harmonic_notch_frequencies = 1;
-    _num_dynamic_harmonic_notches = 1;
-
-    uint8_t num_filters = 0;
-    const bool double_notch = _harmonic_notch_filter.hasOption(HarmonicNotchFilterParams::Options::DoubleNotch);
+    for (auto &notch : harmonic_notches) {
+        if (!notch.params.enabled()) {
+            continue;
+        }
+        notch.calculated_notch_freq_hz[0] = notch.params.center_freq_hz();
+        notch.num_calculated_notch_frequencies = 1;
+        notch.num_dynamic_notches = 1;
 #if APM_BUILD_COPTER_OR_HELI || APM_BUILD_TYPE(APM_BUILD_ArduPlane)
-    if (_harmonic_notch_filter.hasOption(HarmonicNotchFilterParams::Options::DynamicHarmonic)) {
+        if (notch.params.hasOption(HarmonicNotchFilterParams::Options::DynamicHarmonic)) {
```

Comment:
dynamic harmonic is the thing we really only want one of, maybe we should enforce it here